client/src/competition/competition-details-modal/__tests__/competition-details-validation.spec.js

```diff
@@ -0,0 +1,105 @@
+        const expectedResult = {
+          canBeSaved: true,
+          errors: { nameRequired: false, descriptionRequired: false },
+        };
+
+        expect(sut(competitionDetails, initialCompetitionDetails)).toEqual(expectedResult);
+      });
+    });
+
+    describe('when name and description changed', () => {
+      it('should be able to save', () => {
+        const competitionDetails = {
+          name: 'test changed',
+          description: 'some description plus something',
+        };
+        const initialCompetitionDetails = { name: 'test', description: 'some description' };
+        const expectedResult = {
+          canBeSaved: true,
+          errors: { nameRequired: false, descriptionRequired: false },
+        };
+
+        expect(sut(competitionDetails, initialCompetitionDetails)).toEqual(expectedResult);
+      });
+    });
+
+    describe('when name is empty', () => {
+      it('should not be able to save', () => {
+        const competitionDetails = { name: '', description: 'some description' };
+        const initialCompetitionDetails = { name: 'test', description: 'some description' };
+        const expectedResult = {
+          canBeSaved: false,
+          errors: { nameRequired: true, descriptionRequired: false },
+        };
+
+        expect(sut(competitionDetails, initialCompetitionDetails)).toEqual(expectedResult);
+      });
+    });
+
+    describe('when description is empty', () => {
+      it('should not be able to save', () => {
+        const competitionDetails = { name: 'test', description: '' };
+        const initialCompetitionDetails = { name: 'test', description: 'some description' };
+        const expectedResult = {
+          canBeSaved: false,
+          errors: { nameRequired: false, descriptionRequired: true },
+        };
+
+        expect(sut(competitionDetails, initialCompetitionDetails)).toEqual(expectedResult);
+      });
+    });
+
+    describe('when nothing changed', () => {
+      it('should not be able to save', () => {
+        const competitionDetails = { name: 'test', description: 'some description' };
+        const initialCompetitionDetails = { name: 'test', description: 'some description' };
+        const expectedResult = {
+          canBeSaved: false,
+          errors: { nameRequired: false, descriptionRequired: false },
+        };
+
+        expect(sut(competitionDetails, initialCompetitionDetails)).toEqual(expectedResult);
+      });
+    });
+
+    describe('when everything is empty', () => {
+      it('should not be able to save', () => {
+        const competitionDetails = { name: '', description: '' };
+        const initialCompetitionDetails = { name: '', description: '' };
+        const expectedResult = {
+          canBeSaved: false,
+          errors: { nameRequired: true, descriptionRequired: true },
+        };
+
+        expect(sut(competitionDetails, initialCompetitionDetails)).toEqual(expectedResult);
+      });
+    });
+  });
+});
```

Comment:
The tests are written well. It's not a comment, just a tip for the future. All these cases do the same thing, only data is different. For such cases, you could use ```describe.each``` API (https://jestjs.io/docs/en/api#describeeachtable-name-fn-timeout), and just define different test data

Reply:
Cool, thank you